File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
     HashMap<Integer,Integer> map = new HashMap<>();
        int count =0;
        for(int e :nums){
            if(!map.containsKey(e)){
                map.put(e,1);
            }else{
                count+=map.get(e);
                map.put(e,map.getOrDefault(e,0)+1);
            }
        }
        return count;
    }
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    let map = new Map();
    let count =0;
    for(num of nums){
        if(!map.has(num)){
          map.set(num,1);  
        } 
        else{
            count = count + map.get(num);
            map.set(num,map.get(num) +1);
        }
        
    }
        return count;
};
